content/en/docs/tasks/debug/debug-application/debug-running-pod.md

```diff
@@ -718,3 +718,60 @@ Clean up the Pod when you're finished with it:
 ```shell
 kubectl delete pod myapp
 ```
+
+### Custom Profile {#custom-profile}
+
+{{< feature-state for_k8s_version="v1.31" state="beta" >}}
+
+You can define partial container spec as a custom profile in either YAML or JSON format, and apply it using the `--custom` flag.
+
+{{< note >}}
+- Custom profile only supports the modification of the debug container spec. It does not support the modification of the Pod spec of the debug target.
+- Modifications via custom profile is not allowed for certain fields such as command, image, lifecycle, volume devices and container name. In the future, more fields can be added to the disallowed list if required.
+{{< /note >}}
+
+
+First, create a Pod named myapp as an example:
+
+```shell
+kubectl run myapp --image=busybox:1.28 --restart=Never -- sleep 1d
+```
+
+Create a custom profile in a YAML file named `custom-profile.yaml`:
+
+{{% code_sample file="debug/custom-profile.yaml" %}}
+
+You can also use JSON format for the custom profile:
+
+{{% code_sample file="debug/custom-profile.json" %}}
+
+
+Then, debug the Pod using an ephemeral container with the custom profile:
+
+```shell
+kubectl debug -it myapp --image=busybox:1.28 --target=myapp --custom=custom-profile.yaml
```

Comment:
Since the default is `legacy`, would it be better to pass a profile e.g. `--profile=general`?

Reply:
OK, I've addressed